src/pages/workspaces/workspace/Dashboard.js

```diff
@@ -226,6 +227,10 @@ const WorkspaceDashboard = _.flow(
       loadBucketSize()
     } else {
       loadAzureStorage()
+      loadSasToken(workspace.workspace.workspaceId)
+
+      // sas tokens expire after 1 hour
+      setInterval(loadSasToken, Utils.durationToMillis({ minutes: 59 }), workspace.workspace.workspaceId)
```

Comment:
Nit: perhaps make this more like 50 minutes so they would conceivably have time to do something with the token (if they are near the timeout)?

Reply:
Does this interval ever get cleared? As is, it looks like it would keep going after leaving the page. Also, it looks like multiple calls to refresh would result in multiple intervals running at the same time.

Reply:
Good point--we should be passing `signal` in at line 314. @nawatts, would that be sufficient, or does the interval also need to be explicitly cleared in a cleanup method?

Reply:
@aherbst-broad you can see an example of how to do this in `AppLauncher.js` (search for usages of `interval`).

Reply:
Ah good call, checkout https://github.com/DataBiosphere/terra-ui/pull/3202/commits/c7e23b55c98872c7f01f21f9d2c1101e21511b3e